apps/one-drive/src/connectors/one-drive/share-point/delete-item-permission.ts

```
@@ -0,0 +1,33 @@
+import { env } from '@/common/env';
+import { MicrosoftError } from '@/common/error';
+
+type DeleteItemPermissionParams = {
+  itemId: string;
+  token: string;
+  siteId: string;
+  driveId: string;
+  permissionId: string;
+};
+
+export const deleteItemPermission = async ({
+  token,
+  siteId,
+  driveId,
+  itemId,
+  permissionId,
+}: DeleteItemPermissionParams): Promise<void> => {
+  const url = new URL(
+    `${env.MICROSOFT_API_URL}/sites/${siteId}/drives/${driveId}/items/${itemId}/permissions/${permissionId}`
+  );
+
+  const response = await fetch(url, {
+    method: 'DELETE',
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+  });
+
+  if (!response.ok) {
```

Do we really want to throw an error when the permission is not found ? Perhaps the user deleted it by itself before this function is called.

---

In this case response will have status 204, it doesn't matter if there was permission on the file or not.
It will try to remove permission from the file, if there was permission or not, it doesn't matter.
So it will fail if file was deleted or token is invalid. I'm not shoure, maybe we need to throw NonRetriableError ?

---

I think if the file is not found we should delete it using `elba.dataProtection.deleteObject({ ids: [fileId] })`.

---

ok, I will update this, if it got 404, will remove it from Elba